### PR TITLE
Support read/write time to storage

### PR DESCRIPTION
**Pull request by Shylock-Hg**

### CODE REVIEW

File: src/codec/RowWriterV2.cpp [hunk 1]
```diff
@@ -740,6 +742,27 @@ WriteResult RowWriterV2::write(ssize_t index, const Date& v) noexcept {
...
+            memcpy(&buf_[offset + 3 * sizeof(int8_t)],
+                   reinterpret_cast<const void*>(&v.microsec),
+                   sizeof(int32_t));
+            break;
+        default:
+            return WriteResult::TYPE_MISMATCH;
+    }
+    if (field->nullable()) {
+        clearNullBit(field->nullFlagPos());
+    }
```
Comment (panda-sheep) [inline src/codec/RowWriterV2.cpp:762]: Good job,
By the way, what is the range of `time`?

Comment (Shylock-Hg) [reply src/codec/RowWriterV2.cpp:762]: 00:00:00.0-23:59:59:999999

**Review by critical27 — APPROVED** [review]
👍

File: src/codec/test/RowReaderV2Test.cpp [hunk 2]
```diff
@@ -290,8 +298,21 @@ TEST(RowReaderV2, encodedData) {
...
+    t.sec = 45;
+    t.microsec = 54321;
     val = reader->getValueByIndex(12);
+    EXPECT_EQ(Value::Type::TIME, val.type());
+    EXPECT_EQ(t, val.getTime());
+    val = reader->getValueByName("time_col");
+    EXPECT_EQ(Value::Type::TIME, val.type());
+    EXPECT_EQ(t, val.getTime());
+
+    // Col 12 -- non-existing column
```
Comment (panda-sheep) [inline src/codec/test/RowReaderV2Test.cpp:314]: Col 13？

**Review by panda-sheep — COMMENTED** [review]
Good job.

File: src/codec/test/RowReaderV2Test.cpp (hunk 2 above)
Comment (Shylock-Hg) [reply src/codec/test/RowReaderV2Test.cpp]: Good catch. Fixed.

**Review by panda-sheep — APPROVED** [review]
Well done!

**Review by darionyaphet — APPROVED** [review]

**Comment by bright-starry-sky** [comment]
Please take look build error. thanks.

**Comment by Shylock-Hg** [comment]
> Please take look build error. thanks.

Compile OOM,,

**Comment by Shylock-Hg** [comment]
> Please take look build error. thanks.

Why close it ?

**Pull request merged**